src/main/java/de/uos/informatik/ko/rcp/geneticAlgo/GeneratePop.java

```diff
@@ -0,0 +1,101 @@
+            scheduled.add(mycandidate);
+            candidates.remove(mycandidate);
+            //delete scheduled activity from predecessor list
+            for (Map.Entry<Integer, HashSet<Integer>> entry : mypredecessors.entrySet()) {
+                entry.getValue().remove(mycandidate);
+            }
+            while (scheduled.size() < instance.n()) {
+                for (Map.Entry<Integer, HashSet<Integer>> entry : mypredecessors.entrySet()) {
+                    if (entry.getValue().isEmpty()&& (!(scheduled.contains(entry.getKey())))) {
+                        candidates.add(entry.getKey());
+                    }
+                }
+                Integer actcandidate = chooseone(candidates, myGenerator);
+                scheduled.add(actcandidate);
+                candidates.remove(actcandidate);
+                // delete scheduled activity from predecessor list
+                for (Map.Entry<Integer, HashSet<Integer>> entry : mypredecessors.entrySet()) {
+                    entry.getValue().remove(actcandidate);
+                }
+            }
+            ArrayList<Integer> resultsList = new ArrayList<Integer>();
+            for (Integer elem: scheduled){
+                resultsList.add(elem);
+            }
+            results.add(resultsList);
+            scheduled.clear();
+            candidates.clear();
+            if (results.size() >= pop_size){
+                break;
+            }
+        }
+        return results;
+    }
+    // chooses randomly one Integer out of the Integer in the given Set, the Random Generator is given
+    public static Integer chooseone( HashSet<Integer> candidates, Random myGenerator){
+        Integer returncanidate;
+        int mycandidate=0;
+        int item = myGenerator.nextInt(candidates.size());
+        int i = 0;
+        for(Integer elem: candidates) {
+            if (i == item) {
+                mycandidate = elem;
+                break;
+            }
+            i++;
+        }
+        returncanidate = mycandidate;
+        return returncanidate;
+    }
+    //Returns a Hashset as an Array
+    //the rows are the lists from left to right
+    public static int[][] ReturnArray(HashSet<ArrayList<Integer>> mySet){
+        int length = mySet.iterator().next().size();
+        int[][] myArray = new int[mySet.size()][length];
+        Iterator<ArrayList<Integer>> myIterator = mySet.iterator();
+        ArrayList<Integer> currentList;
+        for (int i= 0; i< mySet.size(); i++){
+            currentList = myIterator.next();
+            for (int j= 0; j< length; j++){
+                myArray[i][j]= currentList.get(j);
+            }
+        }
+        return myArray;
+    }
+    public static int GetRowCount(HashSet<ArrayList<Integer>> mySet){
+        return mySet.size();
+    }
+
+    public static int GetColumnCount(HashSet<ArrayList<Integer>> mySet){
+        return mySet.iterator().next().size();
+    }
```

Comment:
What are these needed for? And why do they start with a capital letter?

Reply:
You have a point with the capital letter. :)
I talked to Benedikte and Maria and they wanted to implement there algorithm with an array.
So i created these three functions to give them the chance to use arrays instead of the  HashSet<ArrayList<Integer>>.
Of course, they can use the HashSet, too.

Reply:
Aha, makes sense then :thinking: I think that should be something to discuss with the whole group in that case

Reply:
(I will keep this unresolved until https://github.com/leon-richardt/rcpsp-solver/issues/8 is closed.)